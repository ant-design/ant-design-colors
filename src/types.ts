export type Palette = string[] & { primary?: string };

export type PalettesProps = Record<string, Palette>;
