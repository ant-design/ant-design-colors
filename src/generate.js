"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var tinycolor2_1 = __importDefault(require("tinycolor2"));
var hueStep = 2; // 色相阶梯
var saturationStep = 16; // 饱和度阶梯，浅色部分
var saturationStep2 = 5; // 饱和度阶梯，深色部分
var brightnessStep1 = 5; // 亮度阶梯，浅色部分
var brightnessStep2 = 15; // 亮度阶梯，深色部分
var lightColorCount = 5; // 浅色数量，主色上
var darkColorCount = 4; // 深色数量，主色下
;
function getHue(hsv, i, light) {
    var hue;
    // 根据色相不同，色相转向不同
    if (Math.round(hsv.h) >= 60 && Math.round(hsv.h) <= 240) {
        hue = light ? Math.round(hsv.h) - (hueStep * i) : Math.round(hsv.h) + (hueStep * i);
    }
    else {
        hue = light ? Math.round(hsv.h) + (hueStep * i) : Math.round(hsv.h) - (hueStep * i);
    }
    if (hue < 0) {
        hue += 360;
    }
    else if (hue >= 360) {
        hue -= 360;
    }
    return hue;
}
function getSaturation(hsv, i, light) {
    // grey color don't change saturation
    if (hsv.h === 0 && hsv.s === 0) {
        return hsv.s;
    }
    var saturation;
    if (light) {
        saturation = Math.round(hsv.s * 100) - (saturationStep * i);
    }
    else if (i === darkColorCount) {
        saturation = Math.round(hsv.s * 100) + (saturationStep);
    }
    else {
        saturation = Math.round(hsv.s * 100) + (saturationStep2 * i);
    }
    // 边界值修正
    if (saturation > 100) {
        saturation = 100;
    }
    // 第一格的 s 限制在 6-10 之间
    if (light && i === lightColorCount && saturation > 10) {
        saturation = 10;
    }
    if (saturation < 6) {
        saturation = 6;
    }
    return saturation;
}
function getValue(hsv, i, light) {
    if (light) {
        return Math.round(hsv.v * 100) + (brightnessStep1 * i);
    }
    return Math.round(hsv.v * 100) - (brightnessStep2 * i);
}
function generate(color) {
    var patterns = [];
    var pColor = tinycolor2_1.default(color);
    for (var i = lightColorCount; i > 0; i -= 1) {
        var hsv = pColor.toHsv();
        var colorString = tinycolor2_1.default({
            h: getHue(hsv, i, true),
            s: getSaturation(hsv, i, true),
            v: getValue(hsv, i, true),
        }).toHexString();
        patterns.push(colorString);
    }
    patterns.push(pColor.toHexString());
    for (var i = 1; i <= darkColorCount; i += 1) {
        var hsv = pColor.toHsv();
        var colorString = tinycolor2_1.default({
            h: getHue(hsv, i),
            s: getSaturation(hsv, i),
            v: getValue(hsv, i),
        }).toHexString();
        patterns.push(colorString);
    }
    return patterns;
}
exports.default = generate;
