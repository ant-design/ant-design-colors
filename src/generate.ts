import tinycolor from 'tinycolor2';

const hueStep = 2; // 色相阶梯
const saturationStep = 16; // 饱和度阶梯，浅色部分
const saturationStep2 = 5; // 饱和度阶梯，深色部分
const brightnessStep1 = 5; // 亮度阶梯，浅色部分
const brightnessStep2 = 15; // 亮度阶梯，深色部分
const lightColorCount = 5; // 浅色数量，主色上
const darkColorCount = 4; // 深色数量，主色下

interface HsvObject {
  h: number;
  s: number;
  v: number;
}

function getHue(hsv: HsvObject, i: number, light?: boolean): number {
  let hue: number;
  // 根据色相不同，色相转向不同
  if (Math.round(hsv.h) >= 60 && Math.round(hsv.h) <= 240) {
    hue = light ? Math.round(hsv.h) - hueStep * i : Math.round(hsv.h) + hueStep * i;
  } else {
    hue = light ? Math.round(hsv.h) + hueStep * i : Math.round(hsv.h) - hueStep * i;
  }
  if (hue < 0) {
    hue += 360;
  } else if (hue >= 360) {
    hue -= 360;
  }
  return hue;
}

function getSaturation(hsv: HsvObject, i: number, light?: boolean): number {
  // grey color don't change saturation
  if (hsv.h === 0 && hsv.s === 0) {
    return hsv.s;
  }
  let saturation: number;
  if (light) {
    saturation = Math.round(hsv.s * 100) - saturationStep * i;
  } else if (i === darkColorCount) {
    saturation = Math.round(hsv.s * 100) + saturationStep;
  } else {
    saturation = Math.round(hsv.s * 100) + saturationStep2 * i;
  }
  // 边界值修正
  if (saturation > 100) {
    saturation = 100;
  }
  // 第一格的 s 限制在 6-10 之间
  if (light && i === lightColorCount && saturation > 10) {
    saturation = 10;
  }
  if (saturation < 6) {
    saturation = 6;
  }
  return saturation;
}

function getValue(hsv: HsvObject, i: number, light?: boolean): number {
  if (light) {
    return Math.round(hsv.v * 100) + brightnessStep1 * i;
  }
  return Math.round(hsv.v * 100) - brightnessStep2 * i;
}

export default function generate(color: string): string[] {
  const patterns: Array<string> = [];
  const pColor = tinycolor(color);
  for (let i = lightColorCount; i > 0; i -= 1) {
    const hsv = pColor.toHsv();
    const colorString: string = tinycolor({
      h: getHue(hsv, i, true),
      s: getSaturation(hsv, i, true),
      v: getValue(hsv, i, true),
    }).toHexString();
    patterns.push(colorString);
  }
  patterns.push(pColor.toHexString());
  for (let i = 1; i <= darkColorCount; i += 1) {
    const hsv = pColor.toHsv();
    const colorString: string = tinycolor({
      h: getHue(hsv, i),
      s: getSaturation(hsv, i),
      v: getValue(hsv, i),
    }).toHexString();
    patterns.push(colorString);
  }
  return patterns;
}
