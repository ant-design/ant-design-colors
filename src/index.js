"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var generate_1 = __importDefault(require("./generate"));
exports.generate = generate_1.default;
var presetPrimaryColors = {
    red: '#F5222D',
    volcano: '#FA541C',
    orange: '#FA8C16',
    gold: '#FAAD14',
    yellow: '#FADB14',
    lime: '#A0D911',
    green: '#52C41A',
    cyan: '#13C2C2',
    blue: '#1890FF',
    geekblue: '#2F54EB',
    purple: '#722ED1',
    magenta: '#EB2F96',
    grey: '#666666',
};
exports.presetPrimaryColors = presetPrimaryColors;
var presetPalettes = {};
exports.presetPalettes = presetPalettes;
Object.keys(presetPrimaryColors).forEach(function (key) {
    presetPalettes[key] = generate_1.default(presetPrimaryColors[key]);
});
