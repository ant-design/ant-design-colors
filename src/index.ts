export { default as generate } from './generate';
export * from './presets'
export * from './types'
