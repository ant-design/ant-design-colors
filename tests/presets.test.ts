import * as presets from '../src/presets'

test('presets', () => {
    expect(presets).toMatchSnapshot()
})