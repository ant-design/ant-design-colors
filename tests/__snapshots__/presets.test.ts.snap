// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`presets 1`] = `
Object {
  "blue": Array [
    "#e6f4ff",
    "#bae0ff",
    "#91caff",
    "#69b1ff",
    "#4096ff",
    "#1677ff",
    "#0958d9",
    "#003eb3",
    "#002c8c",
    "#001d66",
  ],
  "blueDark": Array [
    "#111a2c",
    "#112545",
    "#15325b",
    "#15417e",
    "#1554ad",
    "#1668dc",
    "#3c89e8",
    "#65a9f3",
    "#8dc5f8",
    "#b7dcfa",
  ],
  "cyan": Array [
    "#e6fffb",
    "#b5f5ec",
    "#87e8de",
    "#5cdbd3",
    "#36cfc9",
    "#13c2c2",
    "#08979c",
    "#006d75",
    "#00474f",
    "#002329",
  ],
  "cyanDark": Array [
    "#112123",
    "#113536",
    "#144848",
    "#146262",
    "#138585",
    "#13a8a8",
    "#33bcb7",
    "#58d1c9",
    "#84e2d8",
    "#b2f1e8",
  ],
  "geekblue": Array [
    "#f0f5ff",
    "#d6e4ff",
    "#adc6ff",
    "#85a5ff",
    "#597ef7",
    "#2f54eb",
    "#1d39c4",
    "#10239e",
    "#061178",
    "#030852",
  ],
  "geekblueDark": Array [
    "#131629",
    "#161d40",
    "#1c2755",
    "#203175",
    "#263ea0",
    "#2b4acb",
    "#5273e0",
    "#7f9ef3",
    "#a8c1f8",
    "#d2e0fa",
  ],
  "gold": Array [
    "#fffbe6",
    "#fff1b8",
    "#ffe58f",
    "#ffd666",
    "#ffc53d",
    "#faad14",
    "#d48806",
    "#ad6800",
    "#874d00",
    "#613400",
  ],
  "goldDark": Array [
    "#2b2111",
    "#443111",
    "#594214",
    "#7c5914",
    "#aa7714",
    "#d89614",
    "#e8b339",
    "#f3cc62",
    "#f8df8b",
    "#faedb5",
  ],
  "gray": Array [
    "#a6a6a6",
    "#999999",
    "#8c8c8c",
    "#808080",
    "#737373",
    "#666666",
    "#404040",
    "#1a1a1a",
    "#000000",
    "#000000",
  ],
  "green": Array [
    "#f6ffed",
    "#d9f7be",
    "#b7eb8f",
    "#95de64",
    "#73d13d",
    "#52c41a",
    "#389e0d",
    "#237804",
    "#135200",
    "#092b00",
  ],
  "greenDark": Array [
    "#162312",
    "#1d3712",
    "#274916",
    "#306317",
    "#3c8618",
    "#49aa19",
    "#6abe39",
    "#8fd460",
    "#b2e58b",
    "#d5f2bb",
  ],
  "grey": Array [
    "#a6a6a6",
    "#999999",
    "#8c8c8c",
    "#808080",
    "#737373",
    "#666666",
    "#404040",
    "#1a1a1a",
    "#000000",
    "#000000",
  ],
  "greyDark": Array [
    "#151515",
    "#1f1f1f",
    "#2d2d2d",
    "#393939",
    "#494949",
    "#5a5a5a",
    "#6a6a6a",
    "#7b7b7b",
    "#888888",
    "#969696",
  ],
  "lime": Array [
    "#fcffe6",
    "#f4ffb8",
    "#eaff8f",
    "#d3f261",
    "#bae637",
    "#a0d911",
    "#7cb305",
    "#5b8c00",
    "#3f6600",
    "#254000",
  ],
  "limeDark": Array [
    "#1f2611",
    "#2e3c10",
    "#3e4f13",
    "#536d13",
    "#6f9412",
    "#8bbb11",
    "#a9d134",
    "#c9e75d",
    "#e4f88b",
    "#f0fab5",
  ],
  "magenta": Array [
    "#fff0f6",
    "#ffd6e7",
    "#ffadd2",
    "#ff85c0",
    "#f759ab",
    "#eb2f96",
    "#c41d7f",
    "#9e1068",
    "#780650",
    "#520339",
  ],
  "magentaDark": Array [
    "#291321",
    "#40162f",
    "#551c3b",
    "#75204f",
    "#a02669",
    "#cb2b83",
    "#e0529c",
    "#f37fb7",
    "#f8a8cc",
    "#fad2e3",
  ],
  "orange": Array [
    "#fff7e6",
    "#ffe7ba",
    "#ffd591",
    "#ffc069",
    "#ffa940",
    "#fa8c16",
    "#d46b08",
    "#ad4e00",
    "#873800",
    "#612500",
  ],
  "orangeDark": Array [
    "#2b1d11",
    "#442a11",
    "#593815",
    "#7c4a15",
    "#aa6215",
    "#d87a16",
    "#e89a3c",
    "#f3b765",
    "#f8cf8d",
    "#fae3b7",
  ],
  "presetDarkPalettes": Object {
    "blue": Array [
      "#111a2c",
      "#112545",
      "#15325b",
      "#15417e",
      "#1554ad",
      "#1668dc",
      "#3c89e8",
      "#65a9f3",
      "#8dc5f8",
      "#b7dcfa",
    ],
    "cyan": Array [
      "#112123",
      "#113536",
      "#144848",
      "#146262",
      "#138585",
      "#13a8a8",
      "#33bcb7",
      "#58d1c9",
      "#84e2d8",
      "#b2f1e8",
    ],
    "geekblue": Array [
      "#131629",
      "#161d40",
      "#1c2755",
      "#203175",
      "#263ea0",
      "#2b4acb",
      "#5273e0",
      "#7f9ef3",
      "#a8c1f8",
      "#d2e0fa",
    ],
    "gold": Array [
      "#2b2111",
      "#443111",
      "#594214",
      "#7c5914",
      "#aa7714",
      "#d89614",
      "#e8b339",
      "#f3cc62",
      "#f8df8b",
      "#faedb5",
    ],
    "green": Array [
      "#162312",
      "#1d3712",
      "#274916",
      "#306317",
      "#3c8618",
      "#49aa19",
      "#6abe39",
      "#8fd460",
      "#b2e58b",
      "#d5f2bb",
    ],
    "grey": Array [
      "#151515",
      "#1f1f1f",
      "#2d2d2d",
      "#393939",
      "#494949",
      "#5a5a5a",
      "#6a6a6a",
      "#7b7b7b",
      "#888888",
      "#969696",
    ],
    "lime": Array [
      "#1f2611",
      "#2e3c10",
      "#3e4f13",
      "#536d13",
      "#6f9412",
      "#8bbb11",
      "#a9d134",
      "#c9e75d",
      "#e4f88b",
      "#f0fab5",
    ],
    "magenta": Array [
      "#291321",
      "#40162f",
      "#551c3b",
      "#75204f",
      "#a02669",
      "#cb2b83",
      "#e0529c",
      "#f37fb7",
      "#f8a8cc",
      "#fad2e3",
    ],
    "orange": Array [
      "#2b1d11",
      "#442a11",
      "#593815",
      "#7c4a15",
      "#aa6215",
      "#d87a16",
      "#e89a3c",
      "#f3b765",
      "#f8cf8d",
      "#fae3b7",
    ],
    "purple": Array [
      "#1a1325",
      "#24163a",
      "#301c4d",
      "#3e2069",
      "#51258f",
      "#642ab5",
      "#854eca",
      "#ab7ae0",
      "#cda8f0",
      "#ebd7fa",
    ],
    "red": Array [
      "#2a1215",
      "#431418",
      "#58181c",
      "#791a1f",
      "#a61d24",
      "#d32029",
      "#e84749",
      "#f37370",
      "#f89f9a",
      "#fac8c3",
    ],
    "volcano": Array [
      "#2b1611",
      "#441d12",
      "#592716",
      "#7c3118",
      "#aa3e19",
      "#d84a1b",
      "#e87040",
      "#f3956a",
      "#f8b692",
      "#fad4bc",
    ],
    "yellow": Array [
      "#2b2611",
      "#443b11",
      "#595014",
      "#7c6e14",
      "#aa9514",
      "#d8bd14",
      "#e8d639",
      "#f3ea62",
      "#f8f48b",
      "#fafab5",
    ],
  },
  "presetPalettes": Object {
    "blue": Array [
      "#e6f4ff",
      "#bae0ff",
      "#91caff",
      "#69b1ff",
      "#4096ff",
      "#1677ff",
      "#0958d9",
      "#003eb3",
      "#002c8c",
      "#001d66",
    ],
    "cyan": Array [
      "#e6fffb",
      "#b5f5ec",
      "#87e8de",
      "#5cdbd3",
      "#36cfc9",
      "#13c2c2",
      "#08979c",
      "#006d75",
      "#00474f",
      "#002329",
    ],
    "geekblue": Array [
      "#f0f5ff",
      "#d6e4ff",
      "#adc6ff",
      "#85a5ff",
      "#597ef7",
      "#2f54eb",
      "#1d39c4",
      "#10239e",
      "#061178",
      "#030852",
    ],
    "gold": Array [
      "#fffbe6",
      "#fff1b8",
      "#ffe58f",
      "#ffd666",
      "#ffc53d",
      "#faad14",
      "#d48806",
      "#ad6800",
      "#874d00",
      "#613400",
    ],
    "green": Array [
      "#f6ffed",
      "#d9f7be",
      "#b7eb8f",
      "#95de64",
      "#73d13d",
      "#52c41a",
      "#389e0d",
      "#237804",
      "#135200",
      "#092b00",
    ],
    "grey": Array [
      "#a6a6a6",
      "#999999",
      "#8c8c8c",
      "#808080",
      "#737373",
      "#666666",
      "#404040",
      "#1a1a1a",
      "#000000",
      "#000000",
    ],
    "lime": Array [
      "#fcffe6",
      "#f4ffb8",
      "#eaff8f",
      "#d3f261",
      "#bae637",
      "#a0d911",
      "#7cb305",
      "#5b8c00",
      "#3f6600",
      "#254000",
    ],
    "magenta": Array [
      "#fff0f6",
      "#ffd6e7",
      "#ffadd2",
      "#ff85c0",
      "#f759ab",
      "#eb2f96",
      "#c41d7f",
      "#9e1068",
      "#780650",
      "#520339",
    ],
    "orange": Array [
      "#fff7e6",
      "#ffe7ba",
      "#ffd591",
      "#ffc069",
      "#ffa940",
      "#fa8c16",
      "#d46b08",
      "#ad4e00",
      "#873800",
      "#612500",
    ],
    "purple": Array [
      "#f9f0ff",
      "#efdbff",
      "#d3adf7",
      "#b37feb",
      "#9254de",
      "#722ed1",
      "#531dab",
      "#391085",
      "#22075e",
      "#120338",
    ],
    "red": Array [
      "#fff1f0",
      "#ffccc7",
      "#ffa39e",
      "#ff7875",
      "#ff4d4f",
      "#f5222d",
      "#cf1322",
      "#a8071a",
      "#820014",
      "#5c0011",
    ],
    "volcano": Array [
      "#fff2e8",
      "#ffd8bf",
      "#ffbb96",
      "#ff9c6e",
      "#ff7a45",
      "#fa541c",
      "#d4380d",
      "#ad2102",
      "#871400",
      "#610b00",
    ],
    "yellow": Array [
      "#feffe6",
      "#ffffb8",
      "#fffb8f",
      "#fff566",
      "#ffec3d",
      "#fadb14",
      "#d4b106",
      "#ad8b00",
      "#876800",
      "#614700",
    ],
  },
  "presetPrimaryColors": Object {
    "blue": "#1677FF",
    "cyan": "#13C2C2",
    "geekblue": "#2F54EB",
    "gold": "#FAAD14",
    "green": "#52C41A",
    "grey": "#666666",
    "lime": "#A0D911",
    "magenta": "#EB2F96",
    "orange": "#FA8C16",
    "purple": "#722ED1",
    "red": "#F5222D",
    "volcano": "#FA541C",
    "yellow": "#FADB14",
  },
  "purple": Array [
    "#f9f0ff",
    "#efdbff",
    "#d3adf7",
    "#b37feb",
    "#9254de",
    "#722ed1",
    "#531dab",
    "#391085",
    "#22075e",
    "#120338",
  ],
  "purpleDark": Array [
    "#1a1325",
    "#24163a",
    "#301c4d",
    "#3e2069",
    "#51258f",
    "#642ab5",
    "#854eca",
    "#ab7ae0",
    "#cda8f0",
    "#ebd7fa",
  ],
  "red": Array [
    "#fff1f0",
    "#ffccc7",
    "#ffa39e",
    "#ff7875",
    "#ff4d4f",
    "#f5222d",
    "#cf1322",
    "#a8071a",
    "#820014",
    "#5c0011",
  ],
  "redDark": Array [
    "#2a1215",
    "#431418",
    "#58181c",
    "#791a1f",
    "#a61d24",
    "#d32029",
    "#e84749",
    "#f37370",
    "#f89f9a",
    "#fac8c3",
  ],
  "volcano": Array [
    "#fff2e8",
    "#ffd8bf",
    "#ffbb96",
    "#ff9c6e",
    "#ff7a45",
    "#fa541c",
    "#d4380d",
    "#ad2102",
    "#871400",
    "#610b00",
  ],
  "volcanoDark": Array [
    "#2b1611",
    "#441d12",
    "#592716",
    "#7c3118",
    "#aa3e19",
    "#d84a1b",
    "#e87040",
    "#f3956a",
    "#f8b692",
    "#fad4bc",
  ],
  "yellow": Array [
    "#feffe6",
    "#ffffb8",
    "#fffb8f",
    "#fff566",
    "#ffec3d",
    "#fadb14",
    "#d4b106",
    "#ad8b00",
    "#876800",
    "#614700",
  ],
  "yellowDark": Array [
    "#2b2611",
    "#443b11",
    "#595014",
    "#7c6e14",
    "#aa9514",
    "#d8bd14",
    "#e8d639",
    "#f3ea62",
    "#f8f48b",
    "#fafab5",
  ],
}
`;
